### PR TITLE
Added app deployments with authentication service.

### PR DESCRIPTION
**Pull request by seriva**

### CODE REVIEW

File: core/core/src/ansible/roles/deployments/tasks/applications/auth-service/validation.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+- name: Verify number of instances
```
Comment (toszo) [inline core/core/src/ansible/roles/deployments/tasks/applications/auth-service/validation.yml]: I think it is not valid validation now.

File: core/core/src/ansible/roles/deployments/templates/auth-service/auth-service.yml.j2 [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+      schedulerName: default-scheduler
+      securityContext:
+        fsGroup: 1000
+        runAsNonRoot: true
+        runAsUser: 1000
+      terminationGracePeriodSeconds: 60
+      volumes:
+      - configMap:
+          defaultMode: 365
+          name: {{ auth_service_name }}
```
[the remark this answers is not in the document]
Comment (toszo) [reply core/core/src/ansible/roles/deployments/templates/auth-service/auth-service.yml.j2]: Don't remember exactly but there was some issue with keycloak initialization when there was consistency in using quotes.

File: core/core/src/ansible/roles/deployments/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+
+- name: Include deployments
+  include_tasks: "applications/{{ item.name }}/main.yml"
+  vars:
+    data: "{{ item }}"
+  loop: "{{ kubernetes.deployments }}"
```
Comment (to-bar) [inline core/core/src/ansible/roles/deployments/tasks/main.yml]: It seems there is a bug or strange behavior of Ansible: loop is parsed even when condition is not met.
Workaround to consider is:
```
loop: "{{ kubernetes.deployments if kubernetes.deployments else [] }}"
when:
  - kubernetes.deployments is defined
  - kubernetes.deployments
```

File: core/core/src/ansible/roles/deployments/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  when:
+      - kubernetes.deployments is defined and kubernetes.deployments | length > 0
+
```
Comment (to-bar) [inline core/core/src/ansible/roles/deployments/tasks/main.yml]: How about having just one empty line? ;)

File: core/core/src/ansible/roles/deployments/templates/auth-service/auth-service.yml.j2 [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    stop-embedded-server
+  keycloak.sh: |
+    #!/usr/bin/env bash
+
+    set -eu
+
+    /opt/jboss/keycloak/bin/jboss-cli.sh --file=/scripts/keycloak.cli
+
+    exec /opt/jboss/tools/docker-entrypoint.sh -b 0.0.0.0  -c standalone-ha.xml
+    exit "$?"
```
Comment (to-bar) [inline core/core/src/ansible/roles/deployments/templates/auth-service/auth-service.yml.j2]: $? is exit code (always integer) so I think quotation marks are not needed

File: core/core/src/ansible/roles/master/tasks/registry-secrets.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+- name: Apply docker-registry secrets
+  shell: kubectl --kubeconfig=/home/{{ admin_user.name }}/.kube/config create secret docker-registry {{ item.name }} --docker-server={{ item.server_url }} --docker-username={{ item.username }} --docker-password="{{ item.password }}" --docker-email={{ item.email }} --namespace={{ item.namespace }} --dry-run -o yaml | kubectl apply -f -
+  become_user: "{{ admin_user.name }}"
+  loop: "{{ kubernetes.image_registry_secrets }}"
```
Comment (to-bar) [inline core/core/src/ansible/roles/master/tasks/registry-secrets.yml]: Same as for deployments, workaround is:
`loop: "{{ kubernetes.image_registry_secrets if kubernetes.image_registry_secrets else [] }}"`

File: core/core/src/ansible/roles/deployments/tasks/main.yml (hunk 3 above)
Comment (toszo) [reply core/core/src/ansible/roles/deployments/tasks/main.yml]: Will not be changed.

File: core/core/src/ansible/roles/master/tasks/registry-secrets.yml (hunk 6 above)
Comment (toszo) [reply core/core/src/ansible/roles/master/tasks/registry-secrets.yml]: Will not be changed.

File: core/core/src/ansible/roles/deployments/templates/auth-service/auth-service.yml.j2 (hunk 5 above)
Comment (toszo) [reply core/core/src/ansible/roles/deployments/templates/auth-service/auth-service.yml.j2]: But also are not bad here.

**Pull request merged**